### PR TITLE
feat(toolu-runner): self-documenting CLI help for every command

### PR DESCRIPTION
**Pull request by Falconiere**
## What

Rich `--help` for all 7 subcommands. The clap surface moves from `main.rs` (627 lines, over the 500-line ceiling) into a new `cli.rs` (`main.rs` now 493).

- **Top level**: `long_about` explaining the JIT model, `Examples:` + `Environment:` help footer (all 4 env vars: `TOOLU_RUNNER_TOKEN`, `TOOLU_RUNNER_CLIENT_ID`, `TOOLU_RUNNER_LOG`, `TOOLU_RUNNER_ALLOW_VERBOSE`), `propagate_version`; bare `toolu-runner` now prints full help (exit code still 2 — the one intentional behavior delta).
- **Every arg**: default stated (`--config`, `--work`, `--name`, `--labels`), `value_name` + path/URL hints.
- **register**: token resolution order (flag > env > stored login) + single-use JIT warning (closes IMP-DO-012), github.com + GHES examples.
- **login**: `TOOLU_RUNNER_CLIENT_ID` fallback documented, GHES example. **run**: `--once` clarified. **remove**: `--token` marked reserved, `--force` semantics spelled out.
- clap definition self-check (`Command::debug_assert`) runs at startup in debug builds — bin-only crate has no lib target for a unit test; the shell-out CLI tests exercise it.

## Tests

`cli_test.rs` 3 → 7 tests: all subcommands listed in help, Examples/env-var asserts, per-command `--config` default check. Also fixes a latent bug in the manual `live_login_e2e` test: it passed `login --hostname`, a flag that never existed (hostname is positional).

## Docs

README gains a CLI flags reference table (closes IMP-DO-018) and fixes the wrong `login --hostname` example; CLAUDE.md + CHANGELOG synced; IMP-DO-012/018 marked done in docs/IMPROVEMENTS.md.

## Verification

- `cargo test --workspace` green; `cargo clippy --workspace --all-targets` clean.
- Every `<cmd> --help` screen rendered and eyeballed.
- No `#[arg(env = …)]` introduced — token/client-id precedence semantics untouched.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/toolu-runner/src/cli.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  #[arg(long, value_name = "NAME")]
+  pub(crate) name: Option<String>,
+  /// Comma-separated runner labels (default: self-hosted,<os>,<arch>).
+  #[arg(long, value_name = "LABELS", value_delimiter = ',')]
+  pub(crate) labels: Vec<String>,
+  /// Numeric runner group ID (org registrations).
+  ///
+  /// Group *names* are not supported by the JIT API: a non-numeric value
+  /// logs a warning and falls back to the Default group (ID 1).
+  #[arg(long, default_value = "Default", value_name = "ID")]
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:140]: Fixed in 5f56ebe — `runner_group_id` now maps "Default" (the CLI's own default, case-insensitive) to the Default group silently; the warning fires only for other non-numeric group names. Default registrations no longer hit the warn path.

[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:140]: Fixed in 5f56ebe — help now reads "default: the Default group" and documents that only non-numeric values other than "Default" warn and fall back to ID 1.

File: crates/toolu-runner/src/cli.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  ///
+  /// When the file is absent or unreadable, `watch` falls back to
+  /// browsing the default data dir (~/.toolu-runner) read-only.
+  #[arg(long, value_name = "FILE", value_hint = ValueHint::FilePath)]
+  pub(crate) config: Option<PathBuf>,
+}
+
+/// Default `--config` path: `~/.toolu-runner/config.toml`.
+pub(crate) fn default_config_path() -> PathBuf {
+  shared::paths::expand_tilde(Path::new("~/.toolu-runner/config.toml"))
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:213]: Fixed in 5f56ebe — added `run_without_config_reports_the_default_config_path`: runs the real binary with HOME pointed at a tempdir and asserts the error names `<HOME>/.toolu-runner/config.toml`, exercising `default_config_path` tilde expansion end to end.

File: crates/toolu-runner/src/cli.rs [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    "self-hosted".to_owned(),
+    std::env::consts::OS.to_owned(),
+    std::env::consts::ARCH.to_owned(),
+  ]
+}
+
+/// Derive the credentials path from the config path. The credentials
+/// file lives next to `config.toml` in the same directory so users
+/// can override `--config` and have both files move together.
+pub(crate) fn credentials_path_for(config_path: &Path) -> PathBuf {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:238]: Fixed in 5f56ebe — added `run_reports_missing_credentials_next_to_config`: real binary, tempdir config, asserts the error names `credentials.json` next to the `--config` file. The no-parent edge case is unreachable from the CLI (a `--config` path always has a parent component after clap parsing).

File: crates/toolu-runner/src/cli.rs [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+  name.unwrap_or_else(|| {
+    hostname::get()
+      .ok()
+      .and_then(|h| h.into_string().ok())
+      .unwrap_or_else(|| "toolu-runner".to_owned())
+  })
+}
+
+/// Default `--labels`: `self-hosted,<os>,<arch>`.
+pub(crate) fn default_labels() -> Vec<String> {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:227]: Not adding a direct unit test: toolu-runner is a bin-only crate (no lib target), so integration tests cannot import `default_labels`, and the repo quality gate forbids inline `#[cfg(test)]` modules. The value is only observable through a live `generate-jitconfig` register, which the manual live-e2e harness covers; `register --help` states the default and cli_test asserts that text.

File: crates/toolu-runner/src/cli.rs [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  pub(crate) config: Option<PathBuf>,
+}
+
+/// Default `--config` path: `~/.toolu-runner/config.toml`.
+pub(crate) fn default_config_path() -> PathBuf {
+  shared::paths::expand_tilde(Path::new("~/.toolu-runner/config.toml"))
+}
+
+/// Runner name from `--name`, defaulting to the machine hostname.
+pub(crate) fn runner_name_or_hostname(name: Option<String>) -> String {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:217]: Same constraint as `default_labels`: bin-only crate (no lib target to import from) and the repo gate forbids inline `#[cfg(test)]`. The hostname fallback chain is only observable through a live register; covered by the manual live-e2e harness.

File: crates/toolu-runner/src/cli.rs [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+/// can override `--config` and have both files move together.
+pub(crate) fn credentials_path_for(config_path: &Path) -> PathBuf {
+  config_path.parent().map_or_else(
+    || PathBuf::from("credentials.json"),
+    |p| p.join("credentials.json"),
+  )
+}
+
+/// Work-folder string from `--work`, defaulting to `~/.toolu-runner/_work`.
+pub(crate) fn work_folder_or_default(work: Option<&PathBuf>) -> String {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:246]: Same constraint: bin-only crate + no inline `#[cfg(test)]` allowed by the gate. `~/.toolu-runner/_work` is only sent in the live register POST; `register --help` states the default and cli_test asserts that text.

File: crates/toolu-runner/src/login_cmd.rs [hunk 7]
```diff
@@ -8,36 +8,57 @@
...
 use config::auth_store::{AuthStore, StoredToken};
 use shared::RunnerError;
 use wire::net;
 
-use crate::{DEVICE_CLIENT_ID, default_config_path};
+use crate::cli::default_config_path;
+
+/// github.com OAuth App `client_id` for the device-flow `login`.
+/// Placeholder until the real App is registered.
+const DEVICE_CLIENT_ID: &str = "REPLACE_ME";
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/login_cmd.rs:20]: Intentional and guarded: `cmd_login` fails fast before any network call while the placeholder is in effect, with the exact remedy in the error (pass `--client-id` or set `TOOLU_RUNNER_CLIENT_ID`). Not silent. The real App client_id lands once the OAuth App is registered (documented as a placeholder in CLAUDE.md and the const doc).

File: crates/toolu-runner/src/cli.rs [hunk 8]
```diff
@@ -0,0 +1,259 @@
+//! clap CLI surface for `toolu-runner`: the [`Cli`] parser, the
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:1]: No change requested — the finding itself notes the convention is followed.

File: crates/toolu-runner/src/main.rs [hunk 9]
```diff
@@ -445,16 +311,17 @@ fn build_registration_config(
...
+/// own default value and the canonical name of group 1, so it maps to
+/// `None` silently; any other group *name* is not supported by the JIT
+/// API and is WARNed about so the fallback to Default is not silent.
 fn runner_group_id(group: &str) -> Option<i64> {
   let trimmed = group.trim();
   if let Ok(id) = trimmed.parse::<i64>() {
     return Some(id);
   }
-  if !trimmed.is_empty() {
+  if !trimmed.is_empty() && !trimmed.eq_ignore_ascii_case("default") {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/main.rs:324]: The doc was updated in the same commit that added the special case (5f56ebe). Lines 313–316 of main.rs read: "\"Default\" is the CLI's own default value and the canonical name of group 1, so it maps to `None` silently; any other group *name* is not supported by the JIT API and is WARNed about". The special case is documented; not stale.

File: crates/toolu-runner/src/cli.rs [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+    .unwrap_or_else(|| "~/.toolu-runner/_work".to_owned())
+}
+
+/// clap's own consistency check, run at startup in debug builds only:
+/// panics on conflicting or invalid argument definitions. A bin-only
+/// crate has no lib target for a unit test to import [`Cli`] from, so
+/// the integration tests exercise this through `cargo run` (debug
+/// profile) instead.
+#[cfg(debug_assertions)]
+pub(crate) fn debug_assert_cli() {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:258]: Fixed in b4ee3c4 — added `clap_self_check_runs_at_startup`: runs the dev-profile binary and asserts a clean `--version` exit, which is exactly the signal that `debug_assert_cli` ran and passed (an invalid definition panics before printing). Comment in the test spells out the mechanism and the `--release` caveat (5e34e44).

File: crates/toolu-runner/src/cli.rs [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+pub(crate) fn credentials_path_for(config_path: &Path) -> PathBuf {
+  config_path.parent().map_or_else(
+    || PathBuf::from("credentials.json"),
+    |p| p.join("credentials.json"),
+  )
+}
+
+/// Work-folder string from `--work`, defaulting to `~/.toolu-runner/_work`.
+pub(crate) fn work_folder_or_default(work: Option<&PathBuf>) -> String {
+  work
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/src/cli.rs:247]: Intentional: the literal `~/...` string is stored in config (same as `data_dir: "~/.toolu-runner"` in `build_registration_config`) and expanded at use time — `cmd_run` resolves it via `config::resolve_work_dir`, which is `expand_tilde` (crates/config/src/config.rs:410-412). Function moved verbatim from main.rs in this PR; behavior unchanged.

File: crates/toolu-runner/tests/cli_test.rs [hunk 12]
```diff
@@ -45,6 +79,130 @@ fn register_help_lists_required_flags() {
...
+    let stdout = String::from_utf8_lossy(&output.stdout);
+    assert!(
+      stdout.contains("config.toml"),
+      "{subcommand} --help does not state the --config default: {stdout}"
+    );
+  }
+}
+
+#[test]
+fn run_without_config_reports_the_default_config_path() {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/tests/cli_test.rs:142]: The test does exercise the expansion: it sets HOME to a tempdir and asserts the error names `<tempdir>/.toolu-runner/config.toml` — i.e. `~` was replaced by the overridden HOME. That is the tilde-expansion logic verified end to end through the real binary.

File: crates/toolu-runner/tests/cli_test.rs [hunk 13]
```diff
@@ -45,6 +79,130 @@ fn register_help_lists_required_flags() {
...
+  let expected = home.path().join(".toolu-runner/config.toml");
+  assert!(
+    stderr.contains(&*expected.to_string_lossy()),
+    "error should name the default config path {}: {stderr}",
+    expected.display()
+  );
+}
+
+#[test]
+fn run_reports_missing_credentials_next_to_config() {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/tests/cli_test.rs:161]: The no-parent edge case is unreachable through the binary (clap always yields a path whose `parent()` is `Some`), and a direct unit test is impossible: bin-only crate (no lib target to import from) + the repo gate forbids inline `#[cfg(test)]`. The fallback branch is defensive only.

File: crates/toolu-runner/tests/cli_test.rs [hunk 14]
```diff
@@ -45,6 +79,130 @@ fn register_help_lists_required_flags() {
...
+  let expected = dir.path().join("credentials.json");
+  assert!(
+    stderr.contains(&*expected.to_string_lossy()),
+    "error should name the sibling credentials path {}: {stderr}",
+    expected.display()
+  );
+}
+
+#[test]
+fn clap_self_check_runs_at_startup() {
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/tests/cli_test.rs:187]: Acknowledged in the test itself — its comment states the `--release` caveat verbatim. CI and `cargo test` build the dev profile, where the check runs. Making it observable in release would mean shipping the self-check to production binaries, which defeats `debug_assert`.

File: crates/toolu-runner/tests/live_login_e2e.rs [hunk 15]
```diff
@@ -164,13 +164,7 @@ async fn device_flow_login_stores_token() {
...
   // polls until the user approves, then persists the token.
   let status = Command::new(&harness.binary_path)
-    .args([
-      "login",
-      "--hostname",
-      "github.com",
-      "--config",
-      &config_path,
-    ])
+    .args(["login", "github.com", "--config", &config_path])
```
[the remark this answers is not in the document]
Comment (Falconiere) [reply crates/toolu-runner/tests/live_login_e2e.rs:167]: A `--hostname` flag never existed — `hostname` has been a positional arg since LoginArgs was introduced; the old test line failed with a clap error if run. No script can depend on a flag the binary rejects. That is precisely the bug this commit fixed.
